src/libxtp/ImaginaryAxisIntegration.cc

```diff
@@ -0,0 +1,112 @@
+/*
+ *            Copyright 2009-2020 The VOTCA Development Team
+ *                       (http://www.votca.org)
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License")
+ *
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *              http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <boost/progress.hpp>
+
+#include <votca/tools/constants.h>
+
+#include "votca/xtp/ImaginaryAxisIntegration.h"
+#include "votca/xtp/quadrature_factory.h"
+#include "votca/xtp/threecenter.h"
+
+namespace votca {
+namespace xtp {
+
+ImaginaryAxisIntegration::ImaginaryAxisIntegration(
+    const Eigen::VectorXd& energies, const TCMatrix_gwbse& Mmn)
+    : _energies(energies), _Mmn(Mmn) {}
+
+void ImaginaryAxisIntegration::configure(
+    options opt, const RPA& rpa, const Eigen::MatrixXd& kDielMxInv_zero) {
+  _opt = opt;
+  QuadratureFactory::RegisterAll();
+  _gq = std::unique_ptr<GaussianQuadratureBase>(
+      Quadratures().Create(_opt.quadrature_scheme));
+  _gq->configure(_opt.order);
+
+  CalcDielInvVector(rpa, kDielMxInv_zero);
+}
+
+// This function calculates and stores inverses of the microscopic dielectric
+// matrix in a matrix vector
+void ImaginaryAxisIntegration::CalcDielInvVector(
+    const RPA& rpa, const Eigen::MatrixXd& kDielMxInv_zero) {
+  _dielinv_matrices_r.resize(_gq->Order());
+
+  std::cout << "\n... ... Preparing RPA for Gaussian quadrature along "
+               "imaginary axis with "
+            << _opt.order << " points" << std::endl;
```

Comment:
Is this necessary? If so, we have to add the logger to sigma_base

Reply:
no